scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2878,252 @@ cdef class Rotation:
+
+        When we apply the rotation to ``b``, we get vectors close to ``a``:
+
+        >>> rot.apply(b)
+        array([[0. , 1. , 0. ],
+               [0. , 1. , 1.1],
+               [0. , 1. , 0.9]])
+
+        The error for the first vector is 0, and for the last two the error is
+        magnitude 0.1. The `rssd` is the square root of the sum of the
+        weighted squared errors, and the default weights are all 1, so in this
+        case the `rssd` is calculated as
+        ``sqrt(1 * 0**2 + 1 * 0.1**2 + 1 * (-0.1)**2) = 0.141421356237308``
+
+        >>> rot.apply(b) - a
+        array([[ 0., 0.,  0. ],
+               [ 0., 0.,  0.1],
+               [ 0., 0., -0.1]])
+        >>> rssd
+        0.141421356237308
+
+        The sensitivity matrix for this example is as follows:
+
+        >>> sens
+        array([[0.2, 0. , 0.],
+               [0. , 1.5, 1.],
+               [0. , 1. , 1.]])
+
+        Special case 1: One infinite weight, which reduces to the
+        align-constrain process. Here we find a rotation between primary and
+        secondary vectors that can align exactly:
+
+        >>> a = [[0, 1, 0], [0, 1, 1]]
+        >>> b = [[1, 0, 0], [1, 1, 0]]
+        >>> rot, _ = R.align_vectors(a, b, weights=[np.inf, 1])
+        >>> rot.as_matrix()
+        array([[0., 0., 1.],
+               [1., 0., 0.],
+               [0., 1., 0.]])
+        >>> rot.apply(b)
+        array([[0., 1., 0.],
+               [0., 1., 1.]])
+
+        Here the secondary vectors must be best-fit:
+
+        >>> b = [[1, 0, 0], [1, 2, 0]]
+        >>> rot, _ = R.align_vectors(a, b, weights=[np.inf, 1])
+        >>> rot.as_matrix()
+        array([[0., 0., 1.],
+               [1., 0., 0.],
+               [0., 1., 0.]])
+        >>> rot.apply(b)
+        array([[0., 1., 0.],
+               [0., 1., 2.]])
+
+        Special case 2: Find a minimum rotation between primary vectors only:
+
+        >>> a = [1, 0, 0]
+        >>> b = [0, 1, 0]
+        >>> rot, _ = R.align_vectors(a, b)
+        >>> rot.as_matrix()
+        array([[0., 1., 0.],
+               [-1., 0., 0.],
+               [0., 0., 1.]])
+        >>> rot.apply(b)
+        array([1., 0., 0.])
+        """
+        # Check input vectors
+        a = np.asarray(a, dtype=np.double)
+        b = np.asarray(b, dtype=np.double)
+        if a.shape[-1] != 3:
+            raise ValueError("Expected input `a` to have shape (3,) or "
+                             "(N, 3), got {}".format(a.shape))
+        if b.shape[-1] != 3:
+            raise ValueError("Expected input `b` to have shape (3,) or "
+                             "(N, 3), got {}".format(b.shape))
         if a.shape != b.shape:
             raise ValueError("Expected inputs `a` and `b` to have same shapes"
                              ", got {} and {} respectively.".format(
                                 a.shape, b.shape))
 
+        # Find number of vector pairs, and normalize to unit length
+        if a.shape == (3,):
```

Comment:
An alternative way to handle the cases is to convert the array to 2D form from the start using `np.atleast_2d`. Maybe it will be cleaner.

Reply:
I agree, that's cleaner